app/javascript/components/Greeting.js

```diff
@@ -1,12 +1,22 @@
-import React from "react";
-import { useSelector } from "react-redux";
-import Loading from "./Loading";
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+import { getMessages } from '../redux/messages/messages';
+import './common.css';
 
 const Greeting = () => {
-  const { greeting, isLoading } = useSelector((state) => state.greetings);
+  const dispatch = useDispatch();
+  const { messages } = useSelector((state) => state);
+
+  console.log(messages);
+
+  useEffect(() => {
+    dispatch(getMessages());
+  }, []);
+
   return (
-    <div className='greetings'>
-      {isLoading ? <Loading /> : <h1>Greeting: {greeting}</h1>}
+    <div className="greeting">
+      <p>{messages[Math.floor(Math.random() * messages.length)]}</p>
     </div>
```

Comment:
- [ ] Still, I am getting messages directly from the API in a hash. please provide an exact URL to access the messages or convert the JSON to string so that the user can understand the message. When I run `http://localhost:3000/api/messages` I got the result which I attached below.

### `http://localhost:3000/api/messages`
![Screenshot from 2022-02-25 14-57-17](https://user-images.githubusercontent.com/22434660/155691280-f40859dd-dc01-48d1-86c3-aaf5ffe79f47.png)

### `http://localhost:3000/api/messages/1`
![Screenshot from 2022-02-25 14-57-37](https://user-images.githubusercontent.com/22434660/155691288-cd811ea9-7573-419f-806d-f00d16d257fb.png)

Reply:
ok, I think with my last update I fixed it.

Reply:
Ok, I think it should work now, this is my third attempt of requesting a review.